### ISSUE TITLE
Rescraping workers segfault

### ISSUE THREAD

**Issue by pypt** [berkmancenter/mediacloud#540]
`lib/MediaWords/Job/RescrapeMedia.pm` segfaults making researchers very unhappy.

Supervisor log:

```
2019-01-15 14:16:53,640 [86196]: Connecting to RabbitMQ (PID: 86196, hostname: mcservices4, port: 5673, username: mediacloud)...
2019-01-15 14:16:53,670 [86196]: Starting function 'MediaWords::Job::RescrapeMedia' from 'lib/MediaWords/Job/RescrapeMedia.pm'.
2019-01-15 14:16:53,690 [86196]: Consumer tag: amq.ctag-OvMCssAEpuQcNugv9sFRFw
2019-01-15 14:16:53,690 [86196]: Worker is ready and accepting jobs
2019-01-15 14:16:53,691 [86196]: Starting job ID "cc2876ad-558b-45dd-859a-a0e5216ba1ca.ad2eb98d89133c848c0ada4fc9d206e7283ca35dc9b5bca7cd66c2cf5b311d33"...
2019-01-15 14:16:53,691 [86196]: ========
2019-01-15 14:16:53,691 [86196]: Arguments: $args = {'job_states_id' => 396312,'media_id' => 361045};
2019-01-15 14:16:53,691 [86196]: ========
2019-01-15 14:16:53,691 [86196]: 
2019-01-15 14:16:53,745 MediaWords.AbstractJob: MediaWords::Job::RescrapeMedia state: running
<segfaults here>
```

Core dump:

```
mediacloud@mccore1:~/mediacloud$ ./script/run_in_env.sh bash
mediacloud@mccore1:~/mediacloud$ source ~/.virtualenvs/mediacloud/bin/activate
(mediacloud) mediacloud@mccore1:~/mediacloud$ gdb perl core
GNU gdb (Ubuntu 7.11.1-0ubuntu1~16.5) 7.11.1
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from perl...(no debugging symbols found)...done.
[New LWP 86196]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `perl /home/mediacloud/.perlbrew/libs/perl-system@mediacloud/bin/mjm_worker.pl l'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007f1e00d7cbd0 in PyModule_GetDict () from /usr/lib/x86_64-linux-gnu/libpython3.7m.so.1.0
(gdb) where
#0  0x00007f1e00d7cbd0 in PyModule_GetDict () from /usr/lib/x86_64-linux-gnu/libpython3.7m.so.1.0
#1  0x00007f1e0134f144 in XS_Inline__Python_py_call_function (my_perl=<optimized out>, cv=0x3531500) at Python.xs:241
#2  0x00000000004bd5fa in Perl_pp_entersub ()
#3  0x00000000004b64a6 in Perl_runops_standard ()
#4  0x0000000000443bc9 in perl_run ()
#5  0x000000000041cb2b in main ()
```

Offending line in `Inline::Python`: https://github.com/berkmancenter/mediacloud-inline-python-pm/blob/master/Python.xs#L241

Can't replicate on macOS.

**Comment by pypt** [comment]
Tried a couple of things:

* reproducing on macOS
* good old "add a bunch of prints and see where exactly is it failing" method
* removing Python code includes and replacing them with stubs
* adding `sleep(1)`s here and there (weirdly enough they seemed to have some impact on when does the worker crash)
* ...

Then, at some point, the worker managed to process the offending job with `media_id = 361045` setting it to a failed state and not segfaulting, and the rest of the jobs continued on being processed just fine.

So, against my own principles, I've added the following uncommitted hack on production:

```diff
diff --git a/lib/MediaWords/Job/RescrapeMedia.pm b/lib/MediaWords/Job/RescrapeMedia.pm
index bb00b047f..e2f58befc 100644
--- a/lib/MediaWords/Job/RescrapeMedia.pm
+++ b/lib/MediaWords/Job/RescrapeMedia.pm
@@ -45,6 +45,10 @@ sub run_statefully($$;$)
         die "'media_id' is undefined.";
     }
 
+    if ($media_id == 361045) {
+        die "Job with media_id = 361045 manages to segfault me";
+    }
+
     MediaWords::DBI::Media::Rescrape::rescrape_media( $db, $media_id );
 }
 ```

just to work through the immediate rescraping queue that has built up.

Closing for now, hope the issue won't reappear with other mediums.
